cardano-db/cardano-db.cabal

```diff
@@ -20,6 +20,9 @@ extra-source-files:     CHANGELOG.md
 
 library
   default-language:     Haskell2010
+  default-extensions:
+    NoImplicitPrelude
+    OverloadedStrings
```

Comment:
This seems un-related to the aim of the PR. In general I think defining default extensions in the cabal file is a bad idea. Just about the only one where I think this is not a bad idea is `NoImplicitPrelude` .

Reply:
https://github.com/input-output-hk/cardano-db-sync/pull/527#discussion_r575045248

Reply:
I think that using `Text` is what you want to do, in most cases, so this is something that I used quite a lot. Additionally, the current `cardano-prelude` does not export `String`. So there is that.